app/models/cart_item.rb

```diff
@@ -0,0 +1,9 @@
+# typed: strict
+class CartItem < ApplicationRecord
+  belongs_to :item
+
+  validates :quantity, presence: true, numericality: { only_integer: true, greater_than: 0 }
```

Comment:
Ensure that the 'quantity' field is indexed since it's likely to be queried frequently, especially in the context of a shopping cart where operations like summing quantities are common.